src/headingengine.js

```diff
@@ -30,6 +23,39 @@ export default class HeadingEngine extends Plugin {
 	/**
 	 * @inheritDoc
 	 */
+	constructor( editor ) {
+		super( editor );
+
+		const t = editor.t;
+
+		/**
+		 * A set of default localized labels for `config.heading.formats`.
+		 *
+		 * @readonly
+		 * @protected
+		 * @member {Object} #_localizedFormatLabels
+		 */
+		const labels = this._localizedFormatLabels = {
```

Comment:
User config can be localised together with the default config when they are processed by the `_getLocalizedOptions()` method (currently `_options` property, but I commented that it shouldn't be a property). The default config doesn't have to be localized beforehand.

You're right about the other thing, though. The localized labels cannot be defined in the module scope, because they require the `t()` function. I missed that.